Feature/users

CodyBrewer opened the pull request:
# Description

- [x] Set up Knex to work with local PostgreSQL databases for development and testing environments
- [x] Set up Knex to work with PostgreSQL database hosted on heroku for PosgreSQL
- [x] Create Migration for Users Table
  - Columns: 
    * id - primary key auto-incrementing id
    * name -  VARCHAR 
    * email - VARCHAR 
    * password - VARCHAR
- [x] Seed user data
- [x] User DB helper functions
- [x] Auth enpodints
- [x] Update README documentation
    - [x] Setting up the development and testing databases
    - [x] Instructions on running Knex migrations & seeds
## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update


## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

## How Has This Been Tested?

- [x] Tested registration endpoint manually
- [x] Teesting login endpoint manually

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

CodyBrewer committed:
➕ Database Dependencies

added knex and pg to deps

CodyBrewer committed:
🔧 Knex Config

set up knexfile and database knex config

CodyBrewer committed:
🗃 Create Users Table

knex migration for creating the users table

CodyBrewer committed:
🔧 set up knexfile to use local postgres db

development and testing environment set up to run against local database

CodyBrewer committed:
🗃 Fix create_users_table migration

added missing schema before createTable and dropTableIfExists, add timestamps columns for create_at and updated_at

CodyBrewer committed:
➕ Load .env in index.js

added require('dotenv').config() to index

CodyBrewer committed:
➕ Knex-cleaner

add knex-cleaner to deps to clean up seeds

CodyBrewer committed:
🗃 cleanup seed

cleanup seed to clean up the database as first step when running seeds

CodyBrewer committed:
🗃 Change column name to username

changed name column to username

CodyBrewer committed:
🗃 Seed Users

seed for user data

CodyBrewer committed:
✨ /api/ router

base router for routes set up

CodyBrewer committed:
👌 change test to test against api/status for the server status

CodyBrewer committed:
✨ add router to server

router added to server to handle routes on /api/

CodyBrewer commented:
![](https://github.trello.services/images/mini-trello-icon.png) [feature/users](https://trello.com/c/ExQ9tDZe/102-feature-users)

CodyBrewer committed:
🐛 Fix db export

export db function

CodyBrewer committed:
🐛 postgres expects id not userid

change to default increments()

CodyBrewer committed:
🗃 user controller

user controller with getAllUsers, getUserBy, addUser, updateUser, deleteUser controller functions

CodyBrewer committed:
✨ /api/users & /api/users/:userid

+

CodyBrewer committed:
🗃 Add user profile columns

added columns needed for profile data

CodyBrewer committed:
🔥 removed GET HOST/api/users/:userid endpoint

removed users/:userid route as pkey should not be exposed

CodyBrewer committed:
🐛 Fix can only call endpoint once

setting db('users') to query and then calling query caused this weird bug for some reason.

CodyBrewer committed:
🎨 hash seeded password

slimmed seed down to one user, add hashing password.

CodyBrewer committed:
🗃 require username and email to be unique

chained .unique() on username and email columns

CodyBrewer committed:
🔀 daily merge

CodyBrewer committed:
♻️ addUser

fix date object on created_at & updated_at columns. Now returns the result of getUserBy(registeredUserId)

CodyBrewer committed:
✨ api/auth/register

register endpoint completed

CodyBrewer committed:
✨ api/auth/register return jwt

generateToken based off the registeredUsers id and returns with the response.

CodyBrewer committed:
✨ api/auth/login

login endpoint created, returns username and jwt

CodyBrewer committed:
♻️ fix response on api/auth/register endpoint

returns the username of the registered user and a jwt token

CodyBrewer committed:
✏️ fix "check" import from prettier typo

removed import and fix the if statement

CodyBrewer committed:
🔥 remove profile columns & rename username column

CodyBrewer committed:
♻️ replace username with email

replaced username with email on user route and auth route

CodyBrewer committed:
🔥 remove users route

need for a users/ route does not exist currently

CodyBrewer committed:
📝 getting started with database instructions

dave-vazquez commented on api/controllers/index.js:
```diff
@@ -0,0 +1,5 @@
+const Users = require('./users');
```
I like this pattern of separating controllers and routes into individual directories 👌  I've seen someone create one for middleware also. Noting for future self 🗒️

dave-vazquez committed:
🐛 remove try-catch blocks

removes try-catch block so errors can be caught and managed by request handlers

dave-vazquez committed:
🔥 remove error handling in register end-point

dave-vazquez committed:
➖ removes SSL param

ithis is good to have, unfortunately i don't quite know how to get it to work in deployment

dave-vazquez committed:
✨ runs formatter

dave-vazquez commented:
Awesome job on this PR :tada:
